src/stores/game.js

```diff
@@ -244,6 +244,9 @@ export const useGameStore = defineStore('game', {
       this.players.push(cloneDeep(player));
       this.players.sort((player, opponent) => player.pNum - opponent.pNum);
     },
+    removeSpectator(username) {
+      this.spectatingUsers = this.spectatingUsers.filter((spectator) => spectator !== username);
+    },
```

Comment:
Reworked spectate-leave so that it just sends the user who left and remaining clients can remove them individually, rather than resending the entire socketEvent for a published gamestate.